### PR TITLE
Implementation of Design, Interactions, and Testing

### PR DESCRIPTION
**Pull request by mohamedSabry0**
### Pull Request: Implementation of Design, Interactions, and Testing

**Design:**

- Implemented design guidelines, adhering to specified colors, typography, and layout from the original design idea by Gregoire Vella on Behance.
- Followed the Creative Commons license, providing appropriate credit to the author in the README of the project.

**Interactions:**

1. **Splash Screen:**
   - Created a simple splash screen displaying the app name with links to the sign-up and log-in pages.

2. **Sign Up and Log In Pages:**
   - Implemented user registration with full name, email, and password.
   - Enabled user login using email and password.
   - Restricted access to pages requiring login if the user is not authenticated.

3. **Home Page (Categories Page):**
   - Presented the categories page upon user login, displaying category name, icon, and total transaction amount.
   - Implemented navigation to the transactions page for a selected category by clicking on the category item.
   - Added functionality for creating a new category with a dedicated "Add a new category" button.

4. **Transactions Page:**
   - Displayed a list of transactions for a given category, ordered by the most recent.
   - Provided the total amount for the category at the top.
   - Implemented a "Add a new transaction" button for creating new transactions.
   - Enabled navigation back to the home page with the "Back" button.

5. **"Add a New Category" Page:**
   - Created a form to add a new category with mandatory name and icon fields.
   - Implemented navigation to the home page on successful category creation.
   - Enabled navigation back to the home page with the "Back" button.

6. **"Add a New Transaction" Page:**
   - Designed a form for creating new transactions with mandatory name, amount, and category fields.
   - Implemented navigation to the transactions page for the corresponding category on successful transaction creation.
   - Enabled navigation back to the transactions page with the "Back" button.

**Testing:**

- Implemented comprehensive unit and integration tests for all the critical components of the RoR application.

**Technical Requirements:**

- Utilized Postgres as the database.
- Incorporated Devise for authentication.
- Validated all user input to enhance security.
- Employed the `.erb` view template engine.
- Deployed the project, making it accessible online.

**Database Schema:**

- Reflects the specified structure:
  - User: Name, Associations: Has many groups, Has many transactions
  - Group: Name, Icon, Associations: Belongs to user, Has many transactions
  - Payment (formerly Transaction): Name, Amount, Associations: Belongs to author(user), Has many groups

Your insights and feedback on these implementations, especially regarding design adherence and functionality, are crucial for refining the application.

### CODE REVIEW

File: README.md
Comment (Lucas-Masaba) [inline README.md:1]: - [x] As mentioned in the instructions, please record a video presenting the features of the project you built. It should be a video with a maximum length of 5 minutes.
image

File: README.md [hunk 1]
```diff
@@ -1,24 +1,176 @@
...
+# 📖 About InqezWallet <a name="about-inqezwallet"></a>
 
-* Database initialization
+InqezWallet is a budget tracking app designed to assist users in managing their finances efficiently. The app enables users to track their expenses, and budget goals effectively.
 
-* How to run the test suite
+# Live Demo
 
-* Services (job queues, cache servers, search engines, etc.)
+[Demo](https://inqez-wallet.onrender.com)
```
Comment (Lucas-Masaba) [inline README.md:38]: - [ ] Please make sure you're live demo follows the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=). Currently, it looks like the screenshots below:
image
image
image

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<h1>Categories List</h1>
+<ul>
+  <%= render @groups %>
```
Comment (Lucas-Masaba) [inline app/views/groups/index.html.erb]: - [ ] Please make sure to follow the [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
 - Colors
 - Typography: font face, size and weight.
 - Layout: composition and space between elements
### Your Design
image
### Required
image
  - The "Add a new category button" should be at the bottom of the page with the background colour being `#5fb523` and a white rectangular boundary around it.
  - The listed category should have the exact design and alignment as in the original design.
  - The title "Category List" should be at the top of the page as seen in the original design.

File: app/views/groups/show.html.erb
Comment (Lucas-Masaba) [inline app/views/groups/show.html.erb:1]: - [ ] Please follow the design guidelines shared in the project requirements as I mentioned in my previous comment. Your design currently looks like this:
image

File: app/views/payments/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<%= form_with(model: payment, url: payments_path(def_group: initial_group)) do |f| %>
+  <div class="field">
+    <%= f.label :name %>
+    <%= f.text_field :name, required: true, placeholder: 'Enter name' %>
+  </div>
+  <div class="field">
+    <%= f.label :amount %>
+    <%= f.number_field :amount, required: true, step: "0.01", placeholder: 'Enter amount' %>
+  </div>
```
Comment (Lucas-Masaba) [inline app/views/payments/_form.html.erb:9]: - [ ] Please add a "Categories" field to this form as mentioned in the project requirements.
image

File: spec/rails_helper.rb
Comment (Lucas-Masaba) [inline spec/rails_helper.rb:1]: - [ ] You have some failing test cases. Please make sure all you're written tests are passing.
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @mohamedSabry0  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Descriptive README :scroll: 
- Descriptive pull request message :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/payments/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  <div class="field">
+    <%= f.label :categories %>
+    <% groups.each do |group| %>
+      <div>
+        <%= f.label "group_ids_#{group.id}" do %>
+          <%= f.check_box :group_ids, { multiple: true, checked: group.id == initial_group.id }, group.id, nil %>
+          <%= group.name %>
+        <% end %>
+      </div>
+    <% end %>
```
Comment (mohamedSabry0) [inline app/views/payments/_form.html.erb]: @Lucas-Masaba, 
I appreciate your informative and thorough review.
and thank you for your note. Is this part of the form acceptable as input for related groups?

Comment (Lucas-Masaba) [reply app/views/payments/_form.html.erb]: Hello @mohamedSabry0 , yes, you should add in the form field even if it is a related group as a project requirement.

Comment (mohamedSabry0) [reply app/views/payments/_form.html.erb]: @Lucas-Masaba, 
Thank you for your time and continuous support.
One last clarification, I implemented the input for the group as a group of checkboxes. 
Is it required to be changed to a text field instead?

Comment (mohamedSabry0) [reply app/views/payments/_form.html.erb]: @Lucas-Masaba,
Sorry for any confusion. I will wait for the next review, as it seems I had a problem in styling preventing them from appearing as expected.

Thank you again for your continuous support.

**Review by Reem-lab — APPROVED** [review]
# Hi @mohamedSabry0,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Descriptive PR⭐
- [ ]  No Linters error 🤩
- [ ] Good job working on the controllers💪🏻
- [ ] Amazing job working on the models 🔥🔥
- [ ] Kudos on the huge efforts for the tests 👏🏻✨✨✨✨✨
- [ ] Great for making the views clean⭐
- [ ] Your design is great 🔥
- [ ] Good job for the video presentation 🎉
- [ ] Project deployed well ⭐
- [ ] Professional README 💪🏻
- [ ] Kudos for adding the required validations 🚀
- [ ] Amazing working on the routes 🙏🏻


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**